### ISSUE TITLE
Fermi/LAT butterfly and upper limits for the 1FHL and 2FHL sources

### ISSUE THREAD

**Issue by jjlk** [gammapy/gammapy#856]
Hi @axel, 
There is something weird with the butterfly and th upper limits of the 1FHL and 2FHL catalogues. @cdeil find the reason, it is given in PR https://github.com/gammapy/gammapy/pull/855#issuecomment-275929851. I could have a look in the comming week.

![pks2155](https://cloud.githubusercontent.com/assets/15717586/22406295/0aa7f754-e651-11e6-9194-ef18132714f1.png)

**Comment by adonath** [comment]
@jjlk I'm looking into this issue right now...could you share the code to create the plot?  This would help to verify, that the issue is fixed. Thanks!

**Comment by jjlk** [comment]
I @adonath , here the script I used:
```
import matplotlib.pyplot as plt
from gammapy.catalog import SourceCatalog3FGL, SourceCatalog2FHL, SourceCatalog1FHL
from gammapy.utils.energy import EnergyBounds

import numpy as np

# 2155
name_3fgl = '3FGL J2158.8-3013'
name_2fhl = '2FHL J2158.8-3013'
name_1fhl = '1FHL J2158.8-3013'

# load catalogs
fermi_3fgl = SourceCatalog3FGL()
fermi_2fhl = SourceCatalog2FHL()
fermi_1fhl = SourceCatalog1FHL()

# access crab data by corresponding identifier
src_3fgl = fermi_3fgl[name_3fgl]
src_2fhl = fermi_2fhl[name_2fhl]
src_1fhl = fermi_1fhl[name_1fhl]

# 3FHL
ax = src_3fgl.spectral_model.plot(src_3fgl.energy_range, energy_power=2,
                                  label='Fermi 3FGL', color='r',
                                  flux_unit='erg-1 cm-2 s-1')

# set up an energy array to evaluate the butterfly
emin, emax = src_3fgl.energy_range
energy = EnergyBounds.equal_log_spacing(emin, emax, 100)
butterfly_3fg = src_3fgl.spectrum.butterfly(energy)

butterfly_3fg.plot(src_3fgl.energy_range, ax=ax, energy_power=2, color='r',
                   flux_unit='erg-1 cm-2 s-1')

src_3fgl.flux_points.plot(ax=ax, sed_type='eflux', color='r',
                          flux_unit='erg cm-2 s-1')

# 2FHL
src_2fhl.spectral_model.plot(src_2fhl.energy_range, ax=ax, energy_power=2,
                             c='g', label='Fermi 2FHL', flux_unit='erg-1 cm-2 s-1')

# set up an energy array to evaluate the butterfly
emin, emax = src_2fhl.energy_range
energy = EnergyBounds.equal_log_spacing(emin, emax, 100)
butterfly_2fhl = src_2fhl.spectrum.butterfly(energy)

# plot butterfly and flux points
butterfly_2fhl.plot(src_2fhl.energy_range, ax=ax, energy_power=2, color='g',
                    flux_unit='erg-1 cm-2 s-1')
src_2fhl.flux_points.plot(ax=ax, sed_type='dnde', energy_power=2, color='g',
                          flux_unit='cm-2 s-1 erg-1')

# 1FHL
src_1fhl.spectral_model.plot(src_1fhl.energy_range, ax=ax, energy_power=2,
                             c='b', label='Fermi 1FHL', flux_unit='erg-1 cm-2 s-1')

# set up an energy array to evaluate the butterfly 
emin, emax = src_1fhl.energy_range
energy = EnergyBounds.equal_log_spacing(emin, emax, 100)
butterfly_1fhl = src_1fhl.spectrum.butterfly(energy)

# plot butterfly and flux points
butterfly_1fhl.plot(src_1fhl.energy_range, ax=ax, energy_power=2, color='b',
                    flux_unit='erg-1 cm-2 s-1')
src_1fhl.flux_points.plot(ax=ax, sed_type='dnde', energy_power=2, color='b',
                          flux_unit='cm-2 s-1 erg-1')

ax.set_ylim(5.e-12, 8.e-11)
ax.set_ylabel('dN/dE [erg cm-2 s-1]')
plt.legend(loc=0)
plt.show()
``` 
Thanks!

**Comment by cdeil** [comment]
Fixed by @adonath in  #858
